scipy/optimize/_linprog_ip.py

```diff
@@ -0,0 +1,1940 @@
+"""
+An interior-point method for linear programming.
+"""
+# Author: Matt Haberland
+
+from __future__ import print_function, division, absolute_import
+import numpy as np
+import scipy as sp
+import scipy.sparse as sps
+from warnings import warn
+from scipy.linalg import LinAlgError
+from .optimize import OptimizeResult, OptimizeWarning, _check_unknown_options
+#from scipy.optimize import _check_unknown_options
+from scipy.optimize._remove_redundancy import _remove_redundancy
+
+def _clean_inputs(
+        c,
+        A_ub=None,
+        b_ub=None,
+        A_eq=None,
+        b_eq=None,
+        bounds=None):
+    """
+    Given user inputs for a linear programming problem, return the
+    objective vector, upper bound constraints, equality constraints,
+    and simple bounds in a preferred format.
+
+    Parameters
+    ----------
+    c : array_like
+        Coefficients of the linear objective function to be minimized.
+    A_ub : array_like, optional
+        2-D array which, when matrix-multiplied by x, gives the values of the
+        upper-bound inequality constraints at x.
+    b_ub : array_like, optional
+        1-D array of values representing the upper-bound of each inequality
+        constraint (row) in A_ub.
```

Comment:
@rgommers I found https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt and it says that variable names should be in single backticks. Has the convention changed? 
I'm going to try to use the instructions there to build documentation HTML locally so that I don't have to wait an hour after making changes to see how they turned out : ) I think that will help.